libs/ui/src/components/ArtWithDetails/ArtWithDetails.tsx

```diff
@@ -19,43 +17,85 @@ import {
   CommentList,
 } from '../../components'
 import { CommentFormFieldValues } from '../CommentForm/types'
-
 export type ArtWithDetailsProps = {
   art: Art
   queryKey?: QueryKey
 }
 
 export const ArtWithDetails: FC<ArtWithDetailsProps> = ({ art, queryKey }) => {
+  const { executeRecaptcha } = useReCaptcha()
+
   const { toggleLike, isLiked, isLoading } = useLikeArt(art, queryKey)
   const queryClient = useQueryClient()
 
   const router = useRouter()
   const locale = router.locale as StrapiLocale
 
-  const auth = useAuthContext()
-
-  const artCommentMutation = useArtCommentMutation()
+  const artCommentMutation = useMutation({
+    mutationKey: ['art-comment'],
+    mutationFn: (body: {
+      name?: string
+      content: string
+      email?: string
+      art: number
+      recaptchaToken?: string
+    }) => axios.post('/api/comments', body),
+  })
   const { data } = useArtBySlug(art.slug)
 
   if (!art.comments) {
     art = data as Art
   }
 
-  const handleSendForm = ({ name, content, email }: CommentFormFieldValues) => {
-    if (art?.id) {
+  const handleSendForm = async ({
+    name,
+    content,
+    email,
+  }: CommentFormFieldValues) => {
+    if (!art?.id) return
+
+    try {
+      const recaptchaToken = await executeRecaptcha('comment').catch(error => {
+        console.error(error)
+
+        return undefined
```

Comment:
If this part fails, it is best not to perform comment mutation. Because /api/comment always fails without a valid recaptchaToken.

```suggestion
        throw new Error('Recaptcha failed)
```